File: csharp/demo/demo03a-pass-arg.cs
/*
 * PASSING ARGUMENTS
 * Version A: Using lambdas
 */
using System;
using System.Threading;



class Demo03C : IRunnable
{
    public void run()
    {
        Thread thFoo = new Thread(() => doTask(1, 2, "red"));
        Thread thBar = new Thread(() => doTask(3, 4, "blue"));

        thFoo.Start();
        thBar.Start();
    }


    private void doTask(int a, double b, string c)
    {
        Console.WriteLine($"{a} {b} {c}");
    }
}

File: python/demo03a_pass_arg.py
'''
PASSING ARGUMENTS
Version A: Using the Thread's constructor
'''

import threading



def do_task(a: int, b: float, c: str):
    print(f'{a}  {b}  {c}')



th_foo = threading.Thread(target=do_task, args=(1, 2, 'red'))
th_bar = threading.Thread(target=do_task, args=(3, 4, 'blue'))

th_foo.start()
th_bar.start()
